govuk_chat_evaluation/output_guardrails/evaluate.py

```diff
@@ -67,17 +277,66 @@ def recall(self) -> float:
             zero_division=np.nan,  # type: ignore
         )
 
+    def precision_per_guardrail(self) -> List[float]:
+        """Calculates precision for each guardrail individually."""
+        expected_vectors, actual_vectors = self._expected_actual_vectors
+        if not expected_vectors:  # Avoid calling score with empty lists
+            return [np.nan] * NUM_GUARDRAILS
+        return precision_score(
+            expected_vectors,
+            actual_vectors,
+            average=None,  # type: ignore
+            zero_division=0,  # type: ignore
```

Comment:
we have a convention in the rest of this repo to use nan rather than zero, could we use that here?

Edit: same for recall and f1_score